src/thermo/PlasmaPhase.cpp

```diff
@@ -0,0 +1,236 @@
+        throw CanteraError("PlasmaPhase::setElectronEnergyDistributionType",
+            "Unknown type for electron energy distribution.");
+    }
+}
+
+void PlasmaPhase::setIsotropicElectronEnergyDistribution()
+{
+    m_electronEnergyDist.resize(m_nPoints);
+    double x = m_isotropicShapeFactor;
+    double gamma1 = boost::math::tgamma(3.0 / 2.0 * x);
+    double gamma2 = boost::math::tgamma(5.0 / 2.0 * x);
+    double c1 = x * std::pow(gamma2, 1.5) / std::pow(gamma1, 2.5);
+    double c2 = x * std::pow(gamma2 / gamma1, x);
+    m_electronEnergyDist =
+        c1 * m_electronEnergyLevels.sqrt() /
+        std::pow(m_meanElectronEnergy, 1.5) *
+        (-c2 * (m_electronEnergyLevels /
+        m_meanElectronEnergy).pow(x)).exp();
+    checkElectronEnergyDistribution();
+}
+
+void PlasmaPhase::setElectronTemperature(double Te) {
+    m_electronTemp = Te;
+    m_meanElectronEnergy = 3.0 / 2.0 * electronTemperature() *
+                           Boltzmann / ElectronCharge;
+    updateElectronEnergyDistribution();
+}
+
+void PlasmaPhase::setMeanElectronEnergy(double energy) {
+    m_electronTemp = 2.0 / 3.0 * energy * ElectronCharge / Boltzmann;
+    m_meanElectronEnergy = energy;
+    updateElectronEnergyDistribution();
+}
+
+void PlasmaPhase::setElectronEnergyLevels(const double* levels, size_t length)
+{
+    m_nPoints = length;
+    m_electronEnergyLevels = Eigen::Map<const Eigen::ArrayXd>(levels, length);
+    updateElectronEnergyDistribution();
+}
+
+void PlasmaPhase::checkElectronEnergyDistribution() const
+{
+    if ((m_electronEnergyLevels < 0.0).any()) {
+        throw CanteraError("PlasmaPhase::checkElectronEnergyDistribution",
+            "Values of electron energy levels cannot be negative.");
+    }
+    if ((m_electronEnergyDist < 0.0).any()) {
+        throw CanteraError("PlasmaPhase::checkElectronEnergyDistribution",
+            "Values of electron energy distribution cannot be negative.");
+    }
+    if (m_electronEnergyDist[m_nPoints - 1] > 0.01) {
+        warn_user("PlasmaPhase::checkElectronEnergyDistribution",
+        "The value of the last element of electron energy distribution exceed 0.01. "
+        "This indicates that the value of electron energy level is not high enough "
+        "to contain the isotropic distribution at mean electron energy of "
+        "{} eV", m_meanElectronEnergy);
+    }
+}
+
+void PlasmaPhase::setElectronEnergyDistribution(const double* levels,
+                                                const double* dist,
+                                                size_t length)
+{
+    m_distributionType = "discretized";
+    m_nPoints = length;
+    m_electronEnergyLevels =
+        Eigen::Map<const Eigen::ArrayXd>(levels, length);
+    m_electronEnergyDist =
+        Eigen::Map<const Eigen::VectorXd>(dist, length);
+    checkElectronEnergyDistribution();
+    // calculate mean electron energy and electron temperature
+    Eigen::VectorXd eps52 = m_electronEnergyLevels.pow(5./2.);
```

Comment:
Using `ArrayXd` would be more consistent with the other usage in this class, though it has no practical difference here.